### PR TITLE
Remove unused 'skip_deployment_check'; Prevent confusion on 'deployment_check' with rename

### PR DESCRIPTION
**Pull request by okurz**
* Improve name of method 'deployment_check' to prevent confusion
* t: Remove unused and useless 'skip_deployment_check'

Related progress issue: https://progress.opensuse.org/issues/71500

### CODE REVIEW

File: lib/OpenQA/Schema.pm [hunk 1]
```diff
@@ -75,7 +75,7 @@ sub dsn {
     return $self->storage->connect_info->[0]->{dsn};
 }
 
-sub deployment_check {
+sub deployment {
```
Comment (Martchus) [inline lib/OpenQA/Schema.pm:78]: This function does the actual deployment (and does not return an object related to the deployment). So I would rather call it `deploy`.

File: script/upgradedb [hunk 2]
```diff
@@ -104,7 +101,4 @@ if ($prepare_upgrades) {
     $dh->prepare_upgrade({from_version => $prev_version, to_version => $version});
 }
 
-if ($upgrade_database) {
-    OpenQA::Schema::deployment_check($schema, $force);
-}
-
+OpenQA::Schema::deployment($schema, $force) if $upgrade_database;
```
Comment (Martchus) [inline script/upgradedb:104]: This (and the other occurrences) can be written as `$schema->deploy($force)` which would look much nicer. So if you want to change the style I'd change it to that.

File: lib/OpenQA/Schema.pm (hunk 1 above)
Comment (okurz) [reply lib/OpenQA/Schema.pm:78]: yes, that makes sense

Comment (okurz) [reply lib/OpenQA/Schema.pm]: done

File: script/upgradedb (hunk 2 above)
Comment (okurz) [reply script/upgradedb]: I have done that but in a separate commit on top as the method was not really looking like a method yet.

**Review by Martchus — APPROVED** [review]

**Pull request merged**